controller_manager:
  ros__parameters:
    use_sim_time: false
    update_rate: 100  # Hz
    configure_components_on_start: ["BlackMouthSystem"]

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    front_left_joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController
    
    front_right_joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    back_left_joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController
    
    back_right_joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController


joint_state_broadcaster:
  ros__parameters:
    use_sim_time: false


front_left_joint_trajectory_controller:
  ros__parameters:
    use_sim_time: false
    joints:
      - front_left_hip_roll_joint
      - front_left_hip_pitch_joint
      - front_left_elbow_joint

    command_interfaces:
      - position

    state_interfaces:
      - position

    state_publish_rate: 50.0
    action_monitor_rate: 20.0

    allow_partial_joints_goal: true
    open_loop_control: true
    constraints:
      stopped_velocity_tolerance: 0.01
      goal_time: 0.0  # 0.0 (not checked)
      front_left_hip_roll_joint:
        trajectory: 0.0
        goal: 0.0
      front_left_hip_pitch_joint:
        trajectory: 0.0
        goal: 0.0
      front_left_elbow_joint:
        trajectory: 0.0
        goal: 0.0


front_right_joint_trajectory_controller:
  ros__parameters:
    use_sim_time: false
    joints:
      - front_right_hip_roll_joint
      - front_right_hip_pitch_joint
      - front_right_elbow_joint

    command_interfaces:
      - position

    state_interfaces:
      - position

    state_publish_rate: 50.0
    action_monitor_rate: 20.0

    allow_partial_joints_goal: true
    open_loop_control: true
    constraints:
      stopped_velocity_tolerance: 0.01
      goal_time: 0.0  # 0.0 (not checked)
      front_right_hip_roll_joint:
        trajectory: 0.0
        goal: 0.0
      front_right_hip_pitch_joint:
        trajectory: 0.0
        goal: 0.0
      front_right_elbow_joint:
        trajectory: 0.0
        goal: 0.0


back_left_joint_trajectory_controller:
  ros__parameters:
    use_sim_time: false
    joints:
      - back_left_hip_roll_joint
      - back_left_hip_pitch_joint
      - back_left_elbow_joint

    command_interfaces:
      - position

    state_interfaces:
      - position


    state_publish_rate: 50.0
    action_monitor_rate: 20.0

    allow_partial_joints_goal: true
    open_loop_control: true
    constraints:
      stopped_velocity_tolerance: 0.01
      goal_time: 0.0  # 0.0 (not checked)
      back_left_hip_roll_joint:
        trajectory: 0.0
        goal: 0.0
      back_left_hip_pitch_joint:
        trajectory: 0.0
        goal: 0.0
      back_left_elbow_joint:
        trajectory: 0.0
        goal: 0.0


back_right_joint_trajectory_controller:
  ros__parameters:
    use_sim_time: false
    joints:
      - back_right_hip_roll_joint
      - back_right_hip_pitch_joint
      - back_right_elbow_joint

    command_interfaces:
      - position

    state_interfaces: 
      - position

    state_publish_rate: 50.0
    action_monitor_rate: 20.0

    allow_partial_joints_goal: true
    open_loop_control: true
    constraints:
      stopped_velocity_tolerance: 0.01
      goal_time: 0.0  # 0.0 (not checked)
      back_right_hip_roll_joint:
        trajectory: 0.0
        goal: 0.0
      back_right_hip_pitch_joint:
        trajectory: 0.0
        goal: 0.0
      back_right_elbow_joint:
        trajectory: 0.0
        goal: 0.0
